### PR TITLE
Reliable epson

### LINKED ISSUE #70155: Reliable and trustworthy mirror for Epson artifacts (and maybe others)
**Describe the bug**

Epson seems to be particularly aggressive about removing old versions of their drivers and utilities from their mirrors. This leads to build failures appearing for packages that have been building and working just fine for months, if not longer, when they are suddenly removed. This forces version updates, where maybe other thing would take priority. It also hurts reproducibility.

It would be great if we could figure out a way to host a mirror for such artifacts, ideally ensuring authenticity and integrity. 

I briefly looked into the source RPMs, and unfortunately they are not signed with Epson's PGP key, while the binary artifacts are. This makes it a bit iffy to just host them in some arbitrary github repo. I reached out to their `linux-printer` e-mail address provided in the README, but so far no response. 

If we can't get a signed archive from the vendor, then maybe trusted core members could sign the artifacts establishing some trust that way.

**To Reproduce**
1. package version
2. wait a few months
3. watch build failure due to 404

**Expected behavior**
We should be able to build packages for specific versions as long as we require the version.

**Additional context**

https://github.com/NixOS/nixpkgs/pull/69376
https://github.com/NixOS/nixpkgs/pull/69894#issuecomment-536577580

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: 
  - pkgs.epson-201106w
  - pkgs.epkowa
  - pkgs.epson-workforce-635-nx625-series
  - pkgs.epson-alc1100
  - pkgs.epson-escpr
  - pkgs.epson-escpr2
  - pkgs.epson_201207w
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by wucke13**
###### Motivation for this change
Resolves https://github.com/NixOS/nixpkgs/issues/70155

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @symphorien @artuuge @furrycatherder @nphilou @emanueleperuffo @romildo @jorsn

### CODE REVIEW

File: pkgs/misc/drivers/epson-workforce-635-nx625-series/default.nix [hunk 1]
```diff
@@ -13,7 +13,11 @@ in stdenv.mkDerivation rec {
   version = "1.0.1";
 
   src = fetchurl {
-    url = "https://download.ebz.epson.net/dsc/op/stable/SRPMS/${name}-${version}-1lsb3.2.src.rpm";
+    # NOTE: Don't forget to update the webarchive link too!
+    url = [
+      "https://download.ebz.epson.net/dsc/op/stable/SRPMS/${name}-${version}-1lsb3.2.src.rpm";
```
Comment (furrycatherder) [inline pkgs/misc/drivers/epson-workforce-635-nx625-series/default.nix]: Syntax error here?

Comment (wucke13) [reply pkgs/misc/drivers/epson-workforce-635-nx625-series/default.nix]: Yep. Fixed it, now all should build.

**Comment by jorsn** [comment]
Why not use https for all webarchive urls?

In *epson-escpr2* you specifiy no protocol at all (`url = [ … "web.archive.org/…" ]`), in *epson-escpr* you use https while in ￼*epson-workforce-635-nx625-series* it is only http.

**Review by sphaugh — APPROVED** [review]
LGTM 👍

**Comment by wucke13** [comment]
@jorsn Thanks for the update. I rushed trough doing it and did not look into it well enough. Adjusted accordingly, obviously your suggestion is good!
@furrycatherder Thanks for the feedback!
If you would go and recheck that all the webarchive links indeed yield the needed hashes, that would be good. I did, but maybe I made mistakes there too...

File: pkgs/misc/drivers/epkowa/default.nix [hunk 2]
```diff
@@ -30,7 +30,15 @@ let plugins = {
     version = "2.30.4";
 
     src = fetchurl {
-      url = "https://download2.ebz.epson.net/iscan/plugin/perfection-v330/rpm/x64/iscan-perfection-v330-bundle-${version}.x64.rpm.tar.gz";
+      # To find new versions, visit
+      # http://download.ebz.epson.net/dsc/search/01/search/?OSC=LX and search for
+      # some printer like for instance "WF-7210" to get to the most recent
+      # version.  
+      # NOTE: Don't forget to update the webarchive link too!
+      url = [
```
Comment (symphorien) [inline pkgs/misc/drivers/epkowa/default.nix]: It seems `url` should be plural.

Comment (wucke13) [reply pkgs/misc/drivers/epkowa/default.nix]: Done. Also Replaced all the urls there. Unfortunately, I wasn't able to trigger all of the urls to be cached by webarchive. I guess my ip did hit some limit. Maybe you can trie to get all the files from epkowa cached, by accessing the links and clicking on "save
```
sed 's_\(\s*\)url = "\(.*\)";_\1urls = [\n\1  "\2"\n\1  "https://web.archive.org/web/\2"\n\1];_g' --in-place *.nix
``` 
Did the job

**Review by symphorien — APPROVED** [review]
I checked that the fallback url works for at least one of them. Looks good to me.

**Comment by d-goldin** [comment]
Thanks for doing this!

**Comment by wucke13** [comment]
@Lassulus would you mind checking this?

**Comment by Lassulus** [comment]
hmm, I don't have a printer to test this, but most of the commits just changed the upstream url. I would be happy with a more verbose commit message then "etc". maybe something like, `use additional mirror url in src` ?

**Comment by wucke13** [comment]
Ok, adjusted accordingly.

**Pull request merged**